src/uct/tcp/tcp_sockcm_ep.h

```diff
@@ -0,0 +1,46 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "tcp_listener.h"
+
+
+typedef enum uct_tcp_sockcm_ep_state {
+    UCT_TCP_SOCKCM_EP_ON_SERVER      = UCS_BIT(0), /* ep is on the server side */
+    UCT_TCP_SOCKCM_EP_ON_CLIENT      = UCS_BIT(1), /* ep is on the client side */
+    UCT_TCP_SOCKCM_EP_CONNECTED      = UCS_BIT(2), /* connect()/accept()
+                                                      completed successfully */
+    UCT_TCP_SOCKCM_EP_SENDING        = UCS_BIT(3), /* ep started sending data */
+    UCT_TCP_SOCKCM_EP_DATA_SENT      = UCS_BIT(4), /* ep completed sending the data */
+    UCT_TCP_SOCKCM_EP_DATA_RECEIVING = UCS_BIT(5), /* ep started receiving data */
+    UCT_TCP_SOCKCM_EP_DATA_RECVED    = UCS_BIT(6), /* ep completed receiving the data */
+    UCT_TCP_SOCKCM_EP_DISCONNECTING  = UCS_BIT(7), /* uct_ep_disconnect was
+                                                      called on the ep.
+                                                      this ep is not
+                                                      necessarily disconnected
+                                                      yet */
+    UCT_TCP_SOCKCM_EP_FAILED         = UCS_BIT(8)  /* ep is in error state */
```

Comment:
do we really need all these states? 
i'd add them only as needed to make sure nothing is redundant